salt/tests/unit/modules/test_metalk8s_package_manager_yum.py

```diff
@@ -0,0 +1,196 @@
+                # `rpm -qa <package>` return "" and retcode 0
+                if rpm_qa_outputs is not None:
+                    if isinstance(rpm_qa_outputs, dict):
+                        if rpm_qa_outputs.get(command[2]) is None:
+                            out_kwargs['retcode'] = 1
+                            out_kwargs['stderr'] = 'An error has occured'
+                        else:
+                            out_kwargs['stdout'] = rpm_qa_outputs[command[2]]
+                    else:
+                        out_kwargs['stdout'] = rpm_qa_outputs
+                return utils.cmd_output(**out_kwargs)
+            return None
+
+        salt_dict = {
+            'cmd.run_all': MagicMock(side_effect=_rpm_qa_cmd)
+        }
+        list_dependents_mock = MagicMock(
+            return_value={} if list_dependents is None else list_dependents
+        )
+
+        with patch.dict(metalk8s_package_manager_yum.__salt__, salt_dict), \
+                patch("metalk8s_package_manager_yum._list_dependents",
+                      list_dependents_mock):
+            self.assertEqual(
+                metalk8s_package_manager_yum.list_pkg_dependents(
+                    kwargs.pop("name", "my_package"),
+                    **kwargs
+                ),
+                result
+            )
+
+    @parameterized.expand(
+        param.explicit(kwargs=test_case)
+        for test_case in yaml.safe_load(open(YAML_TESTS_FILE))["check_pkg_availability"]
+    )
+    def test_check_pkg_availability(self, pkgs_info, raise_msg=None,
+                                    ybase_installs=True,
+                                    ybase_process_trans=True):
+        """
+        Tests the return of `check_pkg_availability` function
+        """
+        def _ybase_install(name, *_args, **_kwargs):
+            success = True
+
+            if ybase_installs is False:
+                success = False
+            elif isinstance(ybase_installs, dict):
+                success = ybase_installs.get(name, True)
+            else:
+                return ybase_installs
+
+            if success:
+                return True
+            raise Exception("An error has occurred")
+
+        ybase_mock = MagicMock()
+
+        install_mock = ybase_mock.return_value.install
+        install_mock.side_effect = _ybase_install
+        process_trans_mock = ybase_mock.return_value.processTransaction
+        if ybase_process_trans:
+            process_trans_mock.return_value = True
+        else:
+            process_trans_mock.side_effect = Exception("An error has occurred")
+
+        with patch.dict("sys.modules", {"yum": MagicMock()}):
+            reload(metalk8s_package_manager_yum)
+            with patch("yum.YumBase", ybase_mock), \
+                    patch("yum.Errors.InstallError", Exception), \
+                    patch("yum.Errors.YumDownloadError", Exception), \
+                    patch("yum.Errors.YumRPMCheckError", Exception), \
```

Comment:
Why are we mocking those? Isn't the `MagicMock` used in `sys.modules` sufficient?

Reply:
No, because we want to reach the `except` block so we need to catch Exceptions